### PR TITLE
[docs] Applying the Sphinx material theme

### LINKED ISSUE #2617: [TSC] Evaluate the Sphinx Material theme for read-the-docs
The current Sphinx bootstrap theme does not use local TOCs. The Material theme does show both local and global TOCs. This issue is about testing the Material theme to determine if we should switch to this theme for the read-the-docs builds of the documentation.

### PR DESCRIPTION
**Pull request by gkunz**
As part of the transition from markdown to rst, this PR applies
the material theme which provides a local and global table of content.

Fixes #2617

Signed-off-by: Georg Kunz <georg.kunz@ericsson.com>

### CODE REVIEW

**Review by walterkozlowski — APPROVED** [review]

**Review by ildikov — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -6,12 +6,12 @@ pillow>=7.1.0
 alabaster>=0.7,<0.8,!=0.7.5
 commonmark==0.9.1
 recommonmark==0.6.0
-sphinx<2
+sphinx==2.1.0
```
[the remark this answers is not in the document]
Comment (CsatariGergely) [reply requirements.txt]: Uh oh, I will check how to resolve that conflict. 
We will not need relative-link-corrector once we use native rst.

[the remark this answers is not in the document]
Comment (CsatariGergely) [reply requirements.txt]: Fix proposed here: https://github.com/cntt-n/sphinxcontrib-relative-link-corrector/pull/4

[the remark this answers is not in the document]
Comment (gkunz) [reply requirements.txt]: I noticed that the relative-link-corrector already supported this (with a small bug) in version 0.0.4, but the requirements file for our docs was still pulling in 0.0.3.

[the remark this answers is not in the document]
Comment (CsatariGergely) [reply requirements.txt]: yes, that bug I fix in cntt-n/sphinxcontrib-relative-link-corrector#4 :)

[the remark this answers is not in the document]
Comment (CsatariGergely) [reply requirements.txt]: And 0.0.5 is released now :)

[the remark this answers is not in the document]
Comment (gkunz) [reply requirements.txt]: bumped to 0.0.5

**Review by CsatariGergely — APPROVED** [review]

**Comment by collivier** [comment]
Please see tox + *requirements.txt in https://github.com/cntt-n/CNTT/pull/2741
https://github.com/cntt-n/CNTT/pull/2741/files#diff-d0c24725a770ee24898050e5ccd7d04d272887b30c1635c423c905ac497e8795R8
https://github.com/cntt-n/CNTT/pull/2741/files#diff-52c2b819893b2d6585f2a179023463d0b246ec8ca86bbe00920c772f6375180b

**Review by collivier — CHANGES_REQUESTED** [review]
Please see previous comment

**Comment by pgoyal01** [comment]
@gkunz There is a comment from @collivier and a conflicting file. After resolution, you should request @walterkozlowski to merge. Thanks

**Review by pgoyal01 — APPROVED** [review]

**Review by rihabbanday — APPROVED** [review]

**Comment by gkunz** [comment]
@collivier kind reminder

**Review by pgoyal01 — APPROVED** [review]

**Review by steelescotr — APPROVED** [review]

**Comment by gkunz** [comment]
@collivier kind reminder. Thanks

File: test-requirements.txt [hunk 2]
```diff
@@ -1,8 +1,8 @@
-sphinx!=1.6.6,!=1.6.7,!=2.1.0,!=3.0.0,!=3.4.2,<2 # BSD
+sphinx!=1.6.6,!=1.6.7,!=2.1.0,!=3.0.0,!=3.4.2,<3 # BSD
```
Comment (collivier) [inline test-requirements.txt:1]: Here it's the official abstract deps proposed by OpenStack and I would suggest no to change it.

**Comment by collivier** [comment]
@CsatariGergely @gkunz All doc builds are now broken.

**Pull request merged**